src/Templates/BlockTemplatesCompatibility.php

```diff
@@ -0,0 +1,277 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * BlockTemplatesCompatibility class.
+ *
+ * To bridge the gap on compatibility with PHP hooks and blockified templates.
+ *
+ * @internal
+ */
+class BlockTemplatesCompatibility {
+
+	const LOOP_ITEM_ID = 'product-loop-item';
+
+	/**
+	 * Hoding the supported hooks, their positions and the hooked functions.
+	 *
+	 * @var array $hook_data
+	 */
+	protected $hook_data = array(
```

Comment:
I wonder if we should add the ability for extensions to register something to this array too? This would be useful in the case where extensions create a block to replace the old hook they used previously and want to handle the transition between the two implementations (which presumably Woo can handle for them?).

Reply:
I added `woocommerce_blocks_hooks_compatibility_data` filter.